profile/profile-schema-validation.js

```diff
@@ -12,44 +12,84 @@
 //     clients: string_array,
 //     address: string_array,
 // }
+const { del } = require("express/lib/application");
+const createError = require("http-errors");
 const Joi = require("joi");
+const LawyerObject = {
+  auth: Joi.string().required(),
+  image: Joi.string().required(),
+  name: Joi.string().min(5).max(50).required(),
+  expertise: Joi.array().items(
+    Joi.object({
+      fieldName: Joi.string().required().min(10),
+      yearsOfExpertise: Joi.number().required(),
+    })
+  ),
+  about: Joi.string().required().min(100),
+  cost: Joi.number().required(),
+  rating: Joi.number().max(5).min(0),
+  experience: Joi.array().items(
+    Joi.object({
+      NameOfField: Joi.string().required(),
+      yearsOfExperience: Joi.number().required(),
+    })
+  ),
+  availableTimeSlots: Joi.array().items(Joi.string()),
+  practiceAreas: Joi.array().items(Joi.string()),
+  awards: Joi.array().items(Joi.string()),
+  clients: Joi.array().items(Joi.string()),
+  address: Joi.array().items(Joi.string()),
+};
+const newLawyerProfileSchema = Joi.object(LawyerObject);
+newLawyerProfileSchema.keys(image);
 function createProfileSchema(req, res, next) {
-  const newLawyerProfileSchema = Joi.object({
-    auth: Joi.string().required(),
-    image: Joi.string().required(),
-    name: Joi.string().min(5).max(50).required(),
-    expertise: Joi.array().items(
-      Joi.object({
-        fieldName: Joi.string().required().min(10),
-        yearsOfExpertise: Joi.number().required(),
-      })
-    ),
-    about: Joi.string().required().min(100),
-    cost: Joi.number().required(),
-    rating: Joi.number().max(5).min(0),
-    experience: Joi.array().items(
-      Joi.object({
-        NameOfField: Joi.string().required(),
-        yearsOfExperience: Joi.number().required(),
-      })
-    ),
-    availableTimeSlots: Joi.array().items(Joi.string()),
-    practiceAreas: Joi.array().items(Joi.string()),
-    awards: Joi.array().items(Joi.string()),
-    clients: Joi.array().items(Joi.string()),
-    address: Joi.array().items(Joi.string()),
-  });
   const data = {
     ...req.body,
     auth: req.headers.authorization,
   };
-  const result = Joi.validate(data, newLawyerProfileSchema);
-  if (result) {
+  const result = newLawyerProfileSchema.validate(data);
+  if (!result.error) {
+    next();
+  } else {
+    next(createError(400, "bad request"));
+  }
+}
+function updateProfileSchema(req, res, next) {
+  const ParamsToBeUpdated = req.body;
+  let check = 0;
+  for (let keys in ParamsToBeUpdated) {
```

Comment:
If used earlier advice:
add additional options where we can strip unwanted data from request:
    const options = {
        abortEarly: false, // include all errors
        allowUnknown: true, // ignore unknown props
        stripUnknown: true // remove unknown props
    };
https://jasonwatmore.com/post/2020/07/22/nodejs-express-api-request-schema-validation-with-joi

you would not need to try this check. Unwanted parameters don't affect us.